Jellyfin.Plugin.OpenSubtitles/OpenSubtitleDownloader.cs

```diff
@@ -207,124 +168,143 @@ await Task.Delay(250, cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(id))
             {
-                throw new ArgumentNullException(nameof(id));
+                throw new ArgumentException("Missing param", nameof(id));
             }
 
+            if (_login?.User?.RemainingDownloads <= 0)
+            {
+                if (_limitReset < DateTime.UtcNow)
+                {
+                    _logger.LogDebug("Reset time passed, forcing a new login");
+                    // force login because the limit resets at midnight
+                    _login = null;
+                }
+                else
+                {
+                    throw new OpenApiException("OpenSubtitles download limit reached");
```

Comment:
That doesn't seem like an appropriate exception type.

Reply:
I knew this would be mentioned, I'm not sure what exception type to use... RateLimitExceededException doesn't get logged so the user doesn't know why the plugin is not doing anything :/
Any idea what exception type would be more appropriate?

Reply:
I have a hunch and guess the culprit why the RateLimitException isn't logged is this in the server code:

https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Providers/MediaInfo/SubtitleDownloader.cs#L194-L196

but please feel free to correct me @cvium

Reply:
That checks out, wonder who made that change...

![image](https://user-images.githubusercontent.com/24963659/127754568-be93a2b5-775d-409a-8ec6-15fbde7bdbd2.png)

https://github.com/jellyfin/jellyfin/pull/6366

Reply:
server code aside, wouldn't a simple error log write out and then throwing the `RateLimitExceededException` also work?

Reply:
I've replaced the exceptions with RateLimitExceededException and added an error log message, that will create double logs if that PR gets merged though

Reply:
Thank you for the update!
Indeed that would happen if jellyfin/jellyfin#6366 gets merged. but since this PR and the plugin will probably get a release before that server PR will hit a stable JF version I would argue this is a suitable workaround for now.
However, I guess we should open a issue in this repo to track that we should remove the extra log line once the server PR is included in a release.

@crobibero Sorry for the ping, but may I ask you for your thoughts?

Reply:
It's my opinion that the plugin/provider should handle the rate-limit exceeded message, so I will go ahead and close the linked PR